docs/import/manage_disaster_flexible.js

```diff
@@ -0,0 +1,740 @@
+}
+
+/**
+ * Checks all `columnInfos` for values, concatenates missing ones' labels.
+ * @param {Array<ColumnInfo>} columnInfos
+ * @return {string}
+ */
+function validateColumnArray(columnInfos) {
+  return columnInfos.map(validateColumnPathHasValue)
+      .filter((c) => c)
+      .join(', ');
+}
+
+//         Initialization functions: called once per disaster.
+
+/**
+ * Initializes page: sets up change handlers for building-source radios and
+ * initializes the three {@link ScoreInputType} divs. Calls {@link startPending}
+ * before any work is done, and calls {@link finishPending} after all work
+ * completed, so that all other initialization work can skip calling those
+ * functions, since we are guaranteed to be pending until everything is done.
+ * @param {AssetData} assetData
+ * @return {Promise<void>} Promise that completes when all work is done.
+ */
+async function initializeFlexibleDisaster(assetData) {
+  $('#state-based-disaster-asset-selection-table').hide();
+  $('#flexible-data').show();
+  const {buildingSource} = assetData.flexibleData;
+  if (buildingSource !== null) {
+    switch (buildingSource) {
+      case BuildingSource.BUILDING:
+        onBuildingSourceBuildingsSelected();
+        $('#buildings-source-buildings').prop('checked', true);
+        break;
+      case BuildingSource.POVERTY:
+        onBuildingSourcePovertySelected();
+        $('#buildings-source-poverty').prop('checked', true);
+        break;
+      case BuildingSource.DAMAGE:
+        onBuildingSourceDamageSelected();
+        $('#buildings-source-damage').prop('checked', true);
+        break;
+    }
+  }
+  startPending();
+  try {
+    await Promise.all([
+      initializeGeography(),
+      initializePoverty(),
+      initializeBuildings(),
+    ]);
+  } finally {
+    finishPending();
+  }
+}
+
+/**
+ * Initializes geography div setup. Creates the necessary select element and
+ * list of columns; waits for the list of assets to populate the select; then
+ * waits for the list of columns to populate each column select. This is the
+ * simplest initializer, because geography has no special cascading effects. The
+ * other initializers {@link initializePoverty} and {@link initializeBuildings}
+ * do this as well, but have additional tasks.
+ * @return {Promise<*>} Promise that completes when display is complete
+ */
+async function initializeGeography() {
+  createSelectAndColumns('geography').on('change', onGeographyChange);
+  if (await getAssetsAndSetOptionsForSelect(
+          componentsData.geography.path, false)) {
+    return setInitialColumnValues('geography');
+  }
+}
+
+/**
+ * Initializes poverty-related display. In addition to doing the analog of what
+ * {@link initializeGeography} does, there are two complications: geography div
+ * visibility is based on poverty asset having geometries, and we must
+ * initialize a column select (for the building-count column) in a different
+ * div from the standard poverty div.
```

Comment:
why do you need a different div for the building-count column select

Reply:
Because I think it's clearer to have the building-count column in the building-count source section, not with the poverty section. Do you think it would be better up with the poverty section? There would still have to be special-casing around its visibility, so it's a UX question, not code simplicity, I think.

Reply:
gotcha, I meant why can't it just go with the rest of the building stuff in the same div but this is fine

Reply:
It can't go with the building div because this column is only relevant when poverty asset is used for buildings, and building div is hidden then. Not sure if that's what you were suggesting?